app/ui.js

```diff
@@ -11,1745 +11,1755 @@
-            div.appendChild(document.createTextNode(event.message));
-            msg.appendChild(div);
+        div = document.createElement("div");
+        div.className = 'noVNC_location';
+        text = event.filename + ":" + event.lineno + ":" + event.colno;
+        div.appendChild(document.createTextNode(text));
+        msg.appendChild(div);
 
+        if ((event.error !== undefined) &&
+            (event.error.stack !== undefined)) {
             div = document.createElement("div");
-            div.className = 'noVNC_location';
-            text = event.filename + ":" + event.lineno + ":" + event.colno;
-            div.appendChild(document.createTextNode(text));
+            div.className = 'noVNC_stack';
+            div.appendChild(document.createTextNode(event.error.stack));
             msg.appendChild(div);
-
-            if ((event.error !== undefined) &&
-                (event.error.stack !== undefined)) {
-                div = document.createElement("div");
-                div.className = 'noVNC_stack';
-                div.appendChild(document.createTextNode(event.error.stack));
-                msg.appendChild(div);
-            }
-
-            document.getElementById('noVNC_fallback_error')
-                .classList.add("noVNC_open");
-        } catch (exc) {
-            document.write("noVNC encountered an error.");
         }
-        // Don't return true since this would prevent the error
-        // from being printed to the browser console.
-        return false;
-    });
 
-    var _ = Util.Localisation.get;
-
-    UI = {
-
-        connected: false,
-        desktopName: "",
-
-        resizeTimeout: null,
-        statusTimeout: null,
-        hideKeyboardTimeout: null,
-        idleControlbarTimeout: null,
-        closeControlbarTimeout: null,
-
-        controlbarGrabbed: false,
-        controlbarDrag: false,
-        controlbarMouseDownClientY: 0,
-        controlbarMouseDownOffsetY: 0,
+        document.getElementById('noVNC_fallback_error')
+            .classList.add("noVNC_open");
+    } catch (exc) {
+        document.write("noVNC encountered an error.");
+    }
+    // Don't return true since this would prevent the error
+    // from being printed to the browser console.
+    return false;
+});
+
+const UI = {
+
+    connected: false,
+    desktopName: "",
+
+    resizeTimeout: null,
+    statusTimeout: null,
+    hideKeyboardTimeout: null,
+    idleControlbarTimeout: null,
+    closeControlbarTimeout: null,
+
+    controlbarGrabbed: false,
+    controlbarDrag: false,
+    controlbarMouseDownClientY: 0,
+    controlbarMouseDownOffsetY: 0,
+
+    isSafari: false,
+    rememberedClipSetting: null,
+    lastKeyboardinput: null,
+    defaultKeyboardinputLen: 100,
+
+    inhibit_reconnect: true,
+    reconnect_callback: null,
+    reconnect_password: null,
+
+    prime: function(callback) {
+        if (document.readyState === "interactive" || document.readyState === "complete") {
```

Comment:
Yes, but no -- it's more future-proof to say the states where you know you'll work, as opposed to the state where you won't.  Suppose there's a new readyState added in the future (there probably won't be, because it would break a massive number of things that are written like this) that doesn't have the DOM fully loaded?